distribution/lib/Standard/Database/0.0.0-dev/src/DB_Table.enso

```diff
@@ -1010,46 +1008,45 @@ type DB_Table
-                _ : Expression -> self.evaluate_expression value on_problems
-                _ : DB_Column ->
-                    if Helpers.check_integrity self value then value else
-                        Error.throw (Integrity_Error.Error "Column "+value.name)
-                _ : Constant_Column -> self.make_constant_column value
-                _ : Simple_Expression -> value.evaluate self (set_mode==Set_Mode.Update && as=="") on_problems
-                _ : Vector -> Error.throw (Unsupported_Database_Operation "`Vector` for `set`")
-                _ : Array -> Error.throw (Unsupported_Database_Operation.Error "`Array` for `set`")
-                _ : Range -> Error.throw (Unsupported_Database_Operation.Error "`Range` for `set`")
-                _ : Date_Range -> Error.throw (Unsupported_Database_Operation.Error "`Date_Range` for `set`")
-                _ -> Error.throw (Illegal_Argument.Error "Unsupported type for `DB_Table.set`.")
-
-            ## If `as` was specified, use that. Otherwise, if `value` is a
-              `DB_Column`, use its name. In these two cases, do not make it unique.
-              Otherwise, make it unique. If set_mode is Update, however, do not
-              make it unique.
-            new_column_name = if as != "" then as else
-                if value.is_a DB_Column || set_mode==Set_Mode.Update || set_mode==Set_Mode.Add_Or_Update then resolved.name else unique.make_unique resolved.name
-            renamed = resolved.rename new_column_name
-            renamed.if_not_error <| self.column_naming_helper.check_ambiguity self.column_names renamed.name <|
-                index = self.internal_columns.index_of (c -> c.name == renamed.name)
-                check_add = case set_mode of
-                    Set_Mode.Add_Or_Update -> True
-                    Set_Mode.Add -> if index.is_nothing then True else Error.throw (Existing_Column.Error renamed.name)
-                    Set_Mode.Update -> if index.is_nothing then Error.throw (Missing_Column.Error renamed.name) else True
-                new_table = check_add.if_not_error <|
-                    new_col = renamed.as_internal
-                    new_cols = if index.is_nothing then self.internal_columns + [new_col] else
-                        Vector.new self.column_count i-> if i == index then new_col else self.internal_columns.at i
-                    self.updated_columns new_cols
-
-                problem_builder.report_unique_name_strategy unique
-                problem_builder.attach_problems_after on_problems new_table
+        problem_builder = Problem_Builder.new
+        unique = self.column_naming_helper.create_unique_name_strategy
+        unique.mark_used self.column_names
+
+        resolved = case value of
+            _ : Text -> self.make_constant_column value
+            _ : Expression -> Feature.Set.if_supported_else_throw self.connection.dialect "set" <| self.evaluate_expression value on_problems
```

Comment:
should this be 
```suggestion
            _ : Expression -> Feature.Set.if_supported_else_throw self.connection.dialect "set with expressions" <| self.evaluate_expression value on_problems
```
or something like that?

To avoid suggesting to the user that normal set is not allowed?

Reply:
Done